app/controllers/concerns/api_searchable.rb

```diff
@@ -0,0 +1,19 @@
+# encoding: utf-8
+
+require 'active_support/concern'
+
+module ApiSearchable
+  extend ActiveSupport::Concern
+
+  def search_query
+    params.permit(:q)[:q]
+  end
+
+  def search_params
+    params.permit(:page, :size, :sort, fields: {}, fq: searchable_fields).to_h
```

Comment:
That would allow you to specify a subset of fields that should be searched by default. It's not used right now, and in fact poses a security hole, so I've just removed it in b8f209d